go-controller/pkg/ovn/pods_retry.go

```diff
@@ -0,0 +1,170 @@
+package ovn
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	kapi "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+// iterateRetryPods checks if any outstanding pods have been waiting for 60 seconds of last known failure
+// then tries to re-add them if so
+// updateAll forces all pods to be attempted to be retried regardless of the 1 minute delay
+func (oc *Controller) iterateRetryPods(updateAll bool) {
+	oc.retryPodsLock.Lock()
+	defer oc.retryPodsLock.Unlock()
+	now := time.Now()
+	for key, podEntry := range oc.retryPods {
+		if podEntry.ignore && podEntry.logicalPortCacheEntry == nil {
+			// neither addition nor deletion is being retried
+			continue
+		}
+
+		pod := podEntry.pod
+		podDesc := fmt.Sprintf("[%s/%s/%s]", pod.UID, pod.Namespace, pod.Name)
+		// it could be that the Pod got deleted, but Pod's DeleteFunc has not been called yet, so don't retry creation
+		retryAdd := true
+		kPod, err := oc.watchFactory.GetPod(pod.Namespace, pod.Name)
+		if err != nil && errors.IsNotFound(err) {
```

Comment:
if err != nil && !errors.IsNotFound then retryAdd would still be true, and in that case you would try to addRetry with kPod, which is probably nil. You need to use "pod" variable and only update it to kPod if it was found in the cache.

Reply:
++ good catch stupid me